### PR TITLE
Fix TestServiceAlloc test

### LINKED ISSUE #32028: TestServiceAlloc {master}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-test-go/16560/

Failed: TestServiceAlloc {master}

```
master_test.go:468: unexpected sucess
```

Previous issues for this test: #31850 #31880

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix #32028

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36581)

### CODE REVIEW

File: pkg/master/controller.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ import (
 type Controller struct {
 	NamespaceRegistry namespace.Registry
 	ServiceRegistry   service.Registry
+	ServiceRest       *service.REST
```
Comment (wojtek-t) [inline pkg/master/controller.go]: @smarterclayton @lavalamp This PR is not yet fully finished, but before I fixed unit tests (which will be quite painful), I would like to get your opinion about the approach for fixing it and if you have any better suggestions how to do it.

File: pkg/registry/core/service/ipallocator/controller/repair.go [hunk 2]
```diff
@@ -67,6 +67,7 @@ func NewRepair(interval time.Duration, registry service.Registry, network *net.I
 // RunUntil starts the controller until the provided ch is closed.
 func (c *Repair) RunUntil(ch chan struct{}) {
 	wait.Until(func() {
+		time.Sleep(5*time.Second)
```
Comment (wojtek-t) [inline pkg/registry/core/service/ipallocator/controller/repair.go]: Will be removed - was here to proove my hypothesis and this is the reason of flakes.

File: pkg/master/controller.go (hunk 1 above)
Comment (deads2k) [inline pkg/master/controller.go]: > Also @deads2k - if you have suggestions how to improve it.

So we want to go through the "normal" RESTStorage flow to be sure that an IP gets allocated to us.  Any reason we wouldn't use a normal API client and loopback to the API server itself?

Comment (wojtek-t) [inline pkg/master/controller.go]: Oh yeah - that makes total sense - thanks for suggestion. Will fix this PR

**Comment by deads2k** [comment]
lgtm.  Not sure of 1.5 state right now though.  Am I good to tag it?

**Comment by wojtek-t** [comment]
@deads2k - we allow bug-fixes (and flaky-test fixes) - this belong to both categories, so it's fine to tag it.

**Pull request merged**